methods.py

```diff
@@ -787,9 +787,18 @@ def get_compiler_version(env):
             return None
     else:  # TODO: Implement for MSVC
         return None
-    match = re.search("[0-9]+\.[0-9.]+", version)
+    match = re.search(
+        "(?:(?<=version )|(?<=\) )|(?<=^))"
+        "(?P<major>\d+)"
+        "(?:\.(?P<minor>\d*))?"
+        "(?:\.(?P<patch>\d*))?"
+        "(?:-(?P<metadata1>[0-9a-zA-Z-]*))?"
+        "(?:\+(?P<metadata2>[0-9a-zA-Z-]*))?"
+        "(?: (?P<date>[0-9]{8}|[0-9]{6})(?![0-9a-zA-Z]))?",
```

Comment:
Do we even need to match those if we're not interested in their values? The code in `SConstruct` would be simpler if we didn't match those to then ignore them.

Reply:
But when we need them they are there, so we are not needed to reconstruct the regex (like this time). 
Probably the metadata2 is the rarest (the field starting with +). The only reason I included the metadata2 because the semantic versioning has this. The date can be used for example when only the major version is available and the older versions are not working well.
But if it's not desired I will remove the patch, metadata2 and date.

Reply:
Isn't `metadata1` is the pre-release part according to semver, and `metadata2` is just metadata?

Reply:
@Xrayez Yeah, but the `pre-release` is not fit well for this string: ` 9.3-posix 20200320`. The `posix` part is not indicating a pre-release, it's more like "some more information".

Reply:
I guess it's fine, I don't like having code (especially somewhat complex one) that isn't being used, but it doesn't hurt much either in this specific instance. If we ever switch to Meson all this hacky logic should be replaced by something more thorough (such as compiling a simple C file to retrieve preprocessor constants).